### ISSUE TITLE
rems

### ISSUE THREAD

**Issue by cameronjroe** [elementalui/elemental#123]
Thoughts on using rems instead of pixels in css. Bootstrap 4 has moved over to rems and it's pretty well [supported](http://caniuse.com/#search=rem) out there at this point.

**Comment by jossmac** [comment]
Support has greatly improved since I last checked!

I think it would be a good step forward

**Comment by cameronjroe** [comment]
Definitely! I'd also recommend suggesting setting the html element to a `font-size: 62.5%` for the easier use of rems based on a `10px` base. For example, `1.6rem` would be 16px (set on the body) and all the math would make sense. `2.4rem` = `24px`. Cheers.

**Comment by kof** [comment]
In terms of migration to jss, there will be not much profit to migrate to rem as you will be able to define your root size using the theme even at runtime.

**Comment by cameronjroe** [comment]
@kof How will it not be useful if it's eventually compiling to CSS? rems are scalable vs pxs are static.

**Comment by kof** [comment]
well, whats exactly the real life use case for rem?

**Comment by cameronjroe** [comment]
https://css-tricks.com/theres-more-to-the-css-rem-unit-than-font-sizing/

**Comment by kof** [comment]
I am still wondering who uses font size of html element to recalc dimensions of rem based elements. For me its document based design. In component based design there should be no such a global thing that can influence everything on the page implicitly. 

Especially unclear is why font-size should influence the dimensions.

Having javascript access to all css values we can recalculate all dimensions dynamically and explicit.

**Comment by kof** [comment]
I am following the ideology that components have to be self contained and ideally nothing should influence them from the outside.

**Comment by cameronjroe** [comment]
That's a different point from the use of rems entirely. Setting the html font-size for recalc of a rem is merely to understand the related px value for the rem one would be specifying. If you are specifying a rem value in jss, you would be able to cleary see that you'd be setting the equivalent 16px when setting 1.6. Without this recalc on the html element, you would need to compute the ratio based on a 16px base font-size of the standard html element which is confusing. If jss can do it for you, that's fine, but then you wouldn't see the same ratio in a debugger.

**Comment by cameronjroe** [comment]
`rem` unites are based on the global html element.

**Comment by DenisIzmaylov** [comment]
Be aware, `rem` units will block you to get pixel-perfect markup.
